middlewares.ts

```diff
@@ -23,47 +23,63 @@ export const authorizePls = async (
     res: Response,
     next: NextFunction
 ) => {
-    const authorizationHeader = req.headers.authorization;
-    let token;
+    const authorizationHeader: string | undefined = req.headers.authorization;
+    let token = "";
+    let response: AxiosResponse | undefined = undefined;
+    let url = "";
+
     if (authorizationHeader) {
         token = authorizationHeader.split(" ")[1];
     } else if (req.query.token) {
-        token = req.query.token;
+        token = req.query.token as string;
     }
 
-    if (!token || token.length === 0) {
+    if (token.length === 0) {
```

Comment:
Possible uncaught error here if token is undefined. Can happen if auth header is a one word string. Can be fixed with `?`

```suggestion
    if (token?.length === 0) {
```

Reply:
I didn't think about that case with the auth header. I considered that into the code now.